src/builder-context.c

```diff
@@ -900,9 +900,9 @@ builder_context_enable_rofiles (BuilderContext *self,
           new_action.sa_flags = 0;
           sigaction (SIGHUP, &new_action, NULL);
 
-          sigset (SIGINT, SIG_IGN);
-          sigset (SIGPIPE, SIG_IGN);
-          sigset (SIGSTOP, SIG_IGN);
+          signal (SIGINT, SIG_IGN);
+          signal (SIGPIPE, SIG_IGN);
+          signal (SIGSTOP, SIG_IGN);
```

Comment:
I'm not really familiar with signal APIs but this seems like the wrong direction?

My understanding is `sigset` was the newer, better defined, API to use over `signal`. Maybe a newer API replaces this, or do you have more context?

Reply:
The docs say:

> This API is obsolete: new applications should use the POSIX signal API ([sigaction](https://linux.die.net/man/2/sigaction)(2), [sigprocmask](https://linux.die.net/man/2/sigprocmask)(2), etc.)

So I think one of those APIs should be used.

Reply:
And here we have the deprecated msg from the header :laughing: 

``` 
/* Set the disposition of SIG.  */
extern __sighandler_t sigset (int __sig, __sighandler_t __disp) __THROW
  __attribute_deprecated_msg__
    ("Use the signal and sigprocmask functions instead");
#endif
```

But I like the idea with `sigaction` also better, especially because   we already use it in that scope.